File: src/main/java/com/feiliks/blog/BlogRepository.java
package com.feiliks.blog;

import com.feiliks.common.entities.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.PagingAndSortingRepository;

public interface BlogRepository extends PagingAndSortingRepository<Blog, Long> {

    Blog findBySlug(String slug);

    Blog findBySlugAndOwner(String slug, User user);

    boolean existsBySlug(String slug);

    Iterable<Blog> findPublished();

    Iterable<Blog> findRecentActive();

    Page<Blog> findByOwner(User user, Pageable pageable);

}

File: src/main/kotlin/com/feiliks/blog/repositories/BlogRepository.kt
package com.feiliks.blog.repositories

import com.feiliks.blog.entities.BlogEntity
import com.feiliks.blog.entities.UserEntity
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.PagingAndSortingRepository


interface BlogRepository : PagingAndSortingRepository<BlogEntity, Long> {

    fun findBySlug(slug: String?): BlogEntity?

    fun findBySlugAndOwner(slug: String?, user: UserEntity?): BlogEntity?

    fun existsBySlug(slug: String?): Boolean

    fun findPublished(): Set<BlogEntity>

    fun findRecentActive(): Set<BlogEntity>

    fun findByOwner(user: UserEntity, pageable: Pageable): Page<BlogEntity>

}
